gt-elasticsearch/src/main/java/mil/nga/giat/data/elasticsearch/ElasticDataStoreFactory.java

```diff
@@ -211,17 +221,47 @@ public DataStore createDataStore(RestClient client, RestClient proxyClient, Map<
         return dataStore;
     }
 
-    public RestClient createRestClient(Map<String, Serializable> params) throws IOException {
-        return createRestClient(params, null, null);
+    RestClient getRestClient(Map<String, Serializable> params) throws IOException {
+        return getRestClient(params, null, null);
     }
 
-    private RestClient createRestClient(Map<String, Serializable> params, String user, String password) throws IOException {
+    private RestClient getRestClient(Map<String, Serializable> params, String user, String password) throws IOException {
         final String hostName = getValue(HOSTNAME, params);
-        final String[] hosts = hostName.split(",");
-        final Integer defaultPort = getValue(HOSTPORT, params);
-        final Boolean sslRejectUnauthorized = getValue(SSL_REJECT_UNAUTHORIZED, params);
+        final int defaultPort = getValue(HOSTPORT, params);
         final String adminUser = getValue(USER, params);
+        final boolean sslRejectUnauthorized = getValue(SSL_REJECT_UNAUTHORIZED, params);
+        
         final String type = user == null || adminUser == null || user.equals(adminUser) ? "ADMIN" : "PROXY_USER";
+        final String clientKey = String.format("%s @ %s:%d", user, hostName, defaultPort);
+
+        /* if we already have the client ... just return it so we don't needlessly create the builder */
+        if(clients.containsKey(clientKey))
```

Comment:
Can you include `{}` for consistency? Also include space after `if` for consistency.